presto-main/src/main/java/io/prestosql/server/protocol/QueryResultRows.java

```diff
@@ -201,6 +203,10 @@ private Object getLegacyValue(Object value, Type type)
             if (type instanceof TimestampWithTimeZoneType) {
                 return ((SqlTimestampWithTimeZone) value).roundTo(3);
             }
+
+            if (type instanceof TimeType) {
+                value = ((SqlTime) value).roundTo(3);
```

Comment:
JDBC 339 declares that is `supportsParametricDateTime`, but it was not updated.
Does it work with time(>3) as well?

Reply:
That was an oversight -- and a tricky issue. Supporting that to future-proof the client before the type extended is complicated. Not sure what to do here. We can add another capability for TIME(p) and, later, yet another one for TIME(p) W/ TZ, or we can document the next couple of versions as not being forward compatible with the upcoming changes to TIME(p) and TIME(p) W/ TZ types. 
Given that those types are currently utterly broken, I'm not sure it's such a big deal to take that kind of exception. Thoughts?

Reply:
What is the failure mode? does JDBC 339 can read time values produced by 340+ server at all?
ie is it about data or "just" metadata (various metadata APIs, let's not go into details now)?

i am OK with new capability, and OK with a breaking change for TIME. To me, it's most
important that we make this decision consciously.